### PR TITLE
Update Github actions to Ubuntu 20.04

### PR DESCRIPTION
**Pull request by cydanil**
Ubuntu 20.04 is now available as a preview (using version tag, not `ubuntu-latest`)

### CODE REVIEW

**Comment by maweki** [comment]
Great. Do we want to add `mypy` now or later? I guess we're having too many errors still to make that useful and mandatory for now.

I am really bad in terms of building, CI, and packaging so this might be stupid: Could we have an optional report or something?

**Comment by cydanil** [comment]
Yes, it's possible to get reports, published as a comment in the PR. For example, know codecov does that, but I don't know the details. Did you mean to have the report of mypy?

**Comment by maweki** [comment]
I think publishing all errors as comment would be too much for now. I would love to see the errors in the CI log. Maybe together with the tests, maybe as a seperate run? What do you think?

**Comment by cydanil** [comment]
This now runs! The environment and dependencies are now up to date to run the tests. I'd like a review from @kirienko.

I think the mypy test would be interesting. Would it be okay to do in another PR, or rather do it now?

**Comment by maweki** [comment]
> I think the mypy test would be interesting. Would it be okay to do in another PR, or rather do it now?

Either is fine with me. It just will be helpful to look whether any errors pop up in the stuff we're working on at that moment.

**Comment by cydanil** [comment]
Ok, then I'll look into mypy this evening, and hopefully merge this tonight/tomorrow

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -10,27 +10,29 @@ on:
...
-        pip install flake8 nose
-        pip install --upgrade keyrings.alt
-        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+        python3 -m pip install --upgrade pip
+        pip3 install flake8 nose mypy
+        pip3 install --upgrade keyrings.alt
+        if [ -f requirements.txt ]; then pip3 install -r requirements.txt; fi
     - name: Test with nose
       run: |
+        PATH=$PATH:/home/runner/.local/bin  # Update the PATH, to access pip installed executables
```
Comment (kirienko) [inline .github/workflows/tests.yml]: Why do we need this in PATH? I guess we only need to check the source code.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -10,27 +10,29 @@ on:
...
         xvfb-run -a nosetests -v gourmet/tests/test_*
+    - name: Test with mypy
+      run: |
+        PATH=$PATH:/home/runner/.local/bin  # Update the PATH, to access pip installed executables
```
Comment (kirienko) [inline .github/workflows/tests.yml]: Same here

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -10,27 +10,29 @@ on:
...
+        mypy gourmet
```
Comment (kirienko) [inline .github/workflows/tests.yml]: I would either use `mypy gourmet/` to explicitly show that we check the content of the `gourmet/` folder (what's actually happening now), or say `mypy -p gourmet` to show that we check the package. The latter will exclude additional files like tests, supplementary or dead code if any.

**Comment by kirienko** [comment]
This looks good!
I only have a couple of notes:

1. `mypy` tests are skipped now, and I don't understand why. Maybe that's because `nose` tests failed. I truly don't know.
2. More general question is about `python` version: in Ububtu 20.04 the default py3 version is 3.8, so we don't specify it. However this silent agreement erases all mentions on the version we require. I have nothing against supporting only py3.8+, but we should specify this somewhere (say, in `requirements.txt). But to tell the full truth, I'd prefer to support py3.6 as well. I guess we need to discuss this additionally.

**Comment by maweki** [comment]
>     2\. However this silent agreement erases all mentions on the version we require.

I think our consensus was, we target the earliest distribution where python2 is no longer available by default. I would be surprised to find a distribution that ships python3.6 by default but not python2

**Comment by kirienko** [comment]
@maweki I'm not against `py3.8`, as I said. I only want users to be aware of that fact some way or another. And potentially we also aim to provide Windows and OSX versions.

**Comment by cydanil** [comment]
>     1. `mypy` tests are skipped now, and I don't understand why. Maybe that's because `nose` tests failed. I truly don't know.

It's because `nose` failed indeed, all steps afterwards are skipped. I'm still looking for the right settings to keep going after failure. I'll let you know once I do.

**Comment by kirienko** [comment]
@cydanil alternatively, we can create a separate action for `mypy`.

**Comment by maweki** [comment]
> @cydanil alternatively, we can create a separate action for `mypy`.

I think this is fine as a third "report". I will add missing stubs packages seperately then.

**Comment by cydanil** [comment]
Ok! By switching to `matrix` with `continue-on-error`, it's running fine. Thanks for the hint.

**Review by kirienko — APPROVED** [review]
LGTM

**Comment by cydanil** [comment]
Thanks for the fruitful discussion!

**Pull request merged**